index.d.ts

```diff
@@ -0,0 +1,449 @@
+  header?: () => void;
+  noBorder?: boolean;
+  onStateChange?: () => void;
+  iconProp?: {
+    name?: string;
+    Label?: () => void;
+    size?: number;
+    color?: string;
+  };
+  position?: "top" | "bottom";
+}
+
+type AlertProps = {};
+type AlertButton = {
+  label: string;
+  onPress?: () => void;
+  variant?: typeof BUTTON_VARIANTS[keyof typeof BUTTON_VARIANTS];
+};
+type AlertShowParams = {
+  title?: string;
+  description?: string;
+  isCancelable?: boolean;
+  onDismiss?: () => void;
+  buttons?: Array<AlertButton>;
+};
+
+interface AnimatedImageProps extends RNImageProps {
+  imageHeight?: number;
+  imageWidth?: number;
+  imageUrl: string;
+}
+
+interface AvatarProps extends ViewProps {
+  name?: string;
+  variant?: "small" | "medium";
+  bgColor?: typeof theme.colors.background;
+  fontColor?: typeof theme.colors.font;
+  imageUrl?: string;
+}
+
+interface BadgeProps extends ViewProps {
+  badgeColor?: typeof theme.colors.background;
+  content?: string | number;
+  size?: number;
+  fontSize?: typeof theme.fontSizes;
+  fontFamily?: typeof theme.fonts;
+  color?: typeof theme.colors.font;
+}
+
+interface BottomSheetProps extends ViewProps {
+  showCreateOption?: boolean;
+  CreateItemComponent?: React.ReactNode;
+  showCreateOptionLoader?: boolean;
+  createSearchedOptionLabelStyle?: TextProps;
+  createOptionLabel?: string;
+  onPressCreateOption?: () => void;
+  createSearchedOptionContainerStyle?: TouchableProps;
+  data?: Array<any>;
+  title?: string;
+  hide?: () => void;
+  isVisible?: boolean;
+  onItemPress?: () => void;
+  selectedItemIndex?: number;
+  bg?: typeof theme.colors.background;
+  titleContainerStyle?: ViewProps;
+  titleTextStyle?: TextProps;
+  modalParams?: ModalProps;
+  HeaderComponent?: () => void;
+  ContentRow?: () => void;
+  contentType?: "checkbox" | null;
+  canSearch?: boolean;
+  onDonePress?: () => void;
+}
+
+interface ButtonProps extends TouchableProps {
+  variant?: typeof BUTTON_VARIANTS[keyof typeof BUTTON_VARIANTS];
+  label: string;
+  labelStyle?: TextProps;
+  RightIcon?: () => void;
+  LeftIcon?: () => void;
+  disabled?: boolean;
+  isLoading?: boolean;
+  loadingText?: string;
+  fontFamily?: typeof theme.fonts;
+  color?: typeof theme.colors;
+  fontSize?: typeof theme.fontSizes;
+}
+
+type ButtonGroupProps = {
+  activeColor?: typeof theme.colors[keyof typeof theme.colors];
+  inActiveColor?: typeof theme.colors[keyof typeof theme.colors];
+  buttonItems: Array<string>;
+  onPress: () => void;
+  currentActiveBtn: string;
+  wrapperStyle?: ViewProps;
+  buttonTextStyle?: TextProps;
+  buttonStyle?: TouchableProps;
+};
+
+interface CalendarProps extends RNCalendarProps {
+  selectedDate?: string | typeof dayjs;
+}
+
+interface CardProps extends TouchableOpacityProps, StyleProps {
+  elevation?: number;
+}
+
+type CarouselProps = {
+  itemArray: Array<any>;
+  renderItem: () => void;
+  carouselRef?: React.RefObject<Carousel>;
+  onSnapToItem?: () => void;
```

Comment:
@kamolesh97 search all occurrences of `() => void` and check it returns something or we need to use React.Node if it accepts component.

This returns the number.
```suggestion
  onSnapToItem?: () => number;
```